### ISSUE TITLE
Figure out a cross-compatible solution for run-scripts

### ISSUE THREAD

**Issue by gideonthomas** [mozilla/science.mozilla.org#205]
Right now, we use make files to run complex commands that, for e.g. setup the docker container, run migrations, etc.
For now, this is fine. But it isn't great for running complex commands (often multiple) and is not portable across platforms i.e. Windows.

We should try to figure out a cross-compatible solution for this (if there even is one) or figure out a strategy for getting it to work on multiple platforms even if we have to write separate scripts for them.

cc @simonwex, @Pomax, @alicoding, @gvn, @cadecairos, @alanmoo

**Comment by gvn** [comment]
I'm not sure if this is totally crazy, but could you have a Docker image that's sole purpose is running shell scripts? 

I'm thinking something like a modified [Node image](https://hub.docker.com/_/node/) that can run external scripts and that has the build's folder as an externally mapped mount point so that it can manipulate the files within?

**Comment by alicoding** [comment]
I like the idea, but I think if we have to do that it might be easier and
better off just run the whole docker command than setting a bash docker
container don't you think?

**Comment by gideonthomas** [comment]
An issue with this is that it will limit scripting abilities to only what is inside the docker containers. Often you will need to run a script that "generates code for you" or makes modifications to local code that will need to be committed. These won't be possible with a docker shell process since (I believe) volume mounting will listen for changes to the local fs and not the fs inside the docker container.

**Comment by Pomax** [comment]
If I can make an even simpler suggestion: why do we need one script to rule them all? If the purpose of the script is to get the docker instance up and running, then why would be try and find a single solution when we simply have a `buildthisthing.sh` for unix/linus setups, as well as a `buildthisthing.bat` for windows, and call it a day?

**Comment by Pomax** [comment]
No, but in this case that seems unnecessary, as they're "write once, the end" scripts that we need. It doesn't feel like it's worth over-engineering a solution for a one-time task =)

**Comment by simonwex** [comment]
Pomax's solution here seems pretty damn reasonable. Any argument to just making it so?

**Comment by alicoding** [comment]
We shall make that happen 👍 `.sh` for *Nix and `bat` for Windows.

**Comment by gideonthomas** [comment]
*nix covers only unix :D

**Comment by Pomax** [comment]
@gthomas is that a concern about not enough scripts, or that ali did not write `*n*x`? Because if the first, please clarify =)

**Comment by gideonthomas** [comment]
lol i was pulling ali's leg because of the `*n*x` thing

**Comment by cadecairos** [comment]
If I can't dev on Solaris, I'm gonna rage quit.

**Comment by Pomax** [comment]
@cade you can't dev on Solaris.

**Comment by cadecairos** [comment]
(╯°□°）╯︵ xɐɯoԀ@

**Comment by gvn** [comment]
What about BeOS?

**Comment by cade** [comment]
Like hell I can't!

![solaris](https://cloud.githubusercontent.com/assets/3403/14723486/f5546d8c-07d6-11e6-8d67-3dada23fe87e.gif)

**Comment by Pomax** [comment]
`$> cade.beersOwed++`
